nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1520 @@
+              "4     improve methods  ...                statistical\n",
+              "5  replacement gensim  ...                replacement\n",
+              "6    methods generate  ...                     gensim\n",
+              "7   extraction module  ...                      graph\n",
+              "8           pke comes  ...               great choice\n",
+              "9     pke replacement  ...                 wide range\n",
+              "\n",
+              "[10 rows x 6 columns]"
+            ]
+          },
+          "metadata": {
+            "tags": []
+          }
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ev2r_xMKCPJ5"
+      },
+      "source": [
+        "## 🧑‍🔬 Try it yourself!\n",
+        "\n",
+        "**Task**: \n",
+        "\n",
+        "1. Insert your own text that you would like to extract keywords from\n",
+        "2. Select the desired keyword extraction methods\n",
+        "3. Extract keywords"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "PHbZr2a8CPJ7"
+      },
+      "source": [
+        "# Task 1: Add your own input text to e\n",
+        "text = \"Replace this string rambling on about keyword extraction and how great it is with your own text\"\n",
+        "\n",
+        "# Task 2: Select the desired keyword extraction methods you want to compare\n",
+        "selected_algorithms = ['KeyBERT', 'BART-based', 'TFIDF', 'YAKE', 'TextRank', 'TopicRank']\n",
+        "\n",
+        "# Task 3: Execute this cell to compare the extracted keywords\n",
+        "compare_keyword_extraction_algorithms(text, selection=selected_algorithms)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9wfoRIn1zPQD"
+      },
+      "source": [
+        "To-do: \n",
+        "\n",
+        "* [ ] Update summary\n",
+        "    * Requires data\n",
+        "    * Requires training from scratch\n",
+        "    * Repositories are not frequently updated\n",
+        "\n",
+        "\n",
+        "## Summary\n",
+        "\n",
+        "This notebook gave a brief overview of neural keyword extraction.\n",
+        "\n",
+        "If an annotated dataset is available or if the problem domain is closely related to one of the existing dataset and it's possible to invest some time into training a model yourself, neural keyword extraction **promises great performance**.\n",
+        "\n",
+        "We further presented two approaches that are readily available: **KeyBERT** uses the **similarity** between word and text **embeddings** to find keywords which best describe a text and an **end-to-end neural keyword extraction model** based on BART.\n",
+        "\n",
+        "For now, we recommend to first try `pke` to establish a solid baseline and explore the mentioned extraction methods as an addition.\n",
```

Comment:
Solid recommendation, would put this statement a bit more to the foreground through an 👉 emoji before it